src/main/java/com/hazelcast/eureka/one/StatusChangeStrategy.java

```diff
@@ -0,0 +1,39 @@
+package com.hazelcast.eureka.one;
+
+
+import com.google.common.base.Preconditions;
+import com.netflix.appinfo.ApplicationInfoManager;
+import com.netflix.appinfo.InstanceInfo;
+
+interface StatusChangeStrategy {
```

Comment:
I believe strategy separation provided  what we need.
@tjuchniewicz what do you think?

Reply:
@lazerion Looks great for me!